xdsl/dialects/memref.py

```diff
@@ -587,6 +587,33 @@ def get(
         return Cast.build(operands=[source], result_types=[type])
 
 
+@irdl_op_definition
+class MemorySpaceCast(IRDLOperation):
+    name = "memref.memory_space_cast"
+
+    source: Operand = operand_def(MemRefType[Attribute] | UnrankedMemrefType[Attribute])
+    dest: OpResult = result_def(MemRefType[Attribute] | UnrankedMemrefType[Attribute])
+
+    @staticmethod
+    def get(
+        source: SSAValue | Operation,
+        type: MemRefType[Attribute] | UnrankedMemrefType[Attribute],
+    ):
+        return MemorySpaceCast.build(operands=[source], result_types=[type])
```

Comment:
This seems to be using the old style of creating new operations with `get` and `build`.
I believe new operations are now added with an `__init__` instead e.g.: 
https://github.com/xdslproject/xdsl/blob/a030c55d5e6edee72c0a334618290a9bc03689f7/xdsl/dialects/riscv_scf.py#L47_78

Reply:
Updated this to use new constructor style